stl/inc/xstring

```diff
@@ -164,18 +164,33 @@ struct _Char_traits { // properties of a string or stream element
     }
 
     static _CONSTEXPR20 _Elem* assign(
-        _Out_writes_all_(_Count) _Elem* const _First, size_t _Count, const _Elem _Ch) noexcept
-    /* strengthened */ {
+        _Out_writes_all_(_Count) _Elem* const _First, size_t _Count, const _Elem _Ch) noexcept /* strengthened */ {
         // assign _Count * _Ch to [_First, ...)
-        for (_Elem* _Next = _First; _Count > 0; --_Count, ++_Next) {
-            *_Next = _Ch;
+#ifdef __cpp_lib_constexpr_string
+        if (_STD is_constant_evaluated()) {

```

Comment:
If these changes to `char_traits` are needed for implementability then we need to fix the Standard (1) so it is implementable, and (2) to make it clear that users will need to do the same for their custom character traits class if they want `basic_string<meow, my_traits_class, some_allocator` to work in constant expressions.

Reply:
I've filed #1699 as a reminder.